### PR TITLE
BUG: fix categorical lag for custom index

### PR DESCRIPTION
**Pull request by martinfleis**
While doing #745 we have noticed that categorical lag works only for graphs based on default range index.

This fixes it.

### CODE REVIEW

File: libpysal/graph/tests/test_spatial_lag.py [hunk 1]
```diff
@@ -65,3 +65,9 @@ def test_ties_raise(self):
         with pytest.raises(ValueError, match="There are 2 ties that must be broken"):
             self.yc[3] = "a"  # create ties
             _lag_spatial(self.gc, self.yc, categorical=True)
+
+    def test_categorical_custom_index(self):
+        expceted = np.array(["bar", "foo", "bar", "foo"])
```
Comment (jGaboardi) [inline libpysal/graph/tests/test_spatial_lag.py:70]:
```suggestion
        expected = np.array(["bar", "foo", "bar", "foo"])
```

File: libpysal/graph/tests/test_spatial_lag.py [hunk 2]
```diff
@@ -65,3 +65,9 @@ def test_ties_raise(self):
...
+        np.testing.assert_array_equal(
+            expceted, self.g.lag(["foo", "bar", "foo", "foo"])
```
Comment (jGaboardi) [inline libpysal/graph/tests/test_spatial_lag.py:72]:
```suggestion
            expected, self.g.lag(["foo", "bar", "foo", "foo"])
```

**Comment by jGaboardi** [comment]
@martinfleis Do we require BDFL approval for this PR?

cc @sjsrey

**Review by jGaboardi — APPROVED** [review]

**Comment by martinfleis** [comment]
Probably not. I just tagged @sjsrey as he wrote the code I'm touching here.

**Comment by sjsrey** [comment]
> While doing #745 we have noticed that categorical lag works only for graphs based on default range index.
> 
> This fixes it.

Nice catch.

It reminds me that we might want to surface the indexing options that the graph affords in the migration guide.

**Review by sjsrey — APPROVED** [review]

**Pull request merged**